### PR TITLE
Add support for long metric types

### PR DESCRIPTION
**Pull request by smbecker**
Currently, the metric exporter only reads double-based metrics. This adds support for long-based metrics.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricData.cs [hunk 1]
```diff
@@ -27,12 +27,28 @@ public MetricsData(int version, Metric metric, MetricPoint metricPoint) : base(v
...
                 case MetricType.DoubleGauge:
-                    metricDataPoint = new MetricDataPoint(metric.Name, metricPoint.GetGaugeLastValueDouble());
-                    metricDataPoint.DataPointType = DataPointType.Measurement;
+                    metricDataPoint = new MetricDataPoint(metric.Name, metricPoint.GetGaugeLastValueDouble())
+                    {
+                        DataPointType = DataPointType.Measurement
+                    };
+                    break;
+                case MetricType.LongSum:
+                    metricDataPoint = new MetricDataPoint(metric.Name, metricPoint.GetSumLong())
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricData.cs:42]: maybe add a link to https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/numeric-conversions#implicit-numeric-conversions .

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricHelper.cs [hunk 2]
```diff
@@ -14,22 +14,21 @@ internal class MetricHelper
...
             foreach (var metric in batch)
             {
-                if (metric.MetricType == MetricType.DoubleSum || metric.MetricType == MetricType.DoubleGauge)
+                if (MetricsData.IsSupportedType(metric.MetricType))
                 {
                     foreach (ref readonly var metricPoint in metric.GetMetricPoints())
                     {
-                        telemetryItem = new TelemetryItem(metricPoint.EndTime.UtcDateTime, roleName, roleInstance, instrumentationKey);
-                        telemetryItem.Data = new MonitorBase
+                        telemetryItems.Add(new TelemetryItem(metricPoint.EndTime.UtcDateTime, roleName, roleInstance, instrumentationKey)
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricHelper.cs:24]: while nothing wrong with this refactoring, i'd strongly suggest to keep it separate PR, to make it easy for reviewers to.

**Review by cijothomas — APPROVED** [review]
Might need changelog, tests?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricData.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricData.cs:42]: We have a PR pending in draft state - https://github.com/Azure/azure-sdk-for-net/pull/27544. As the implicit or explicit conversion from long to double loses a precision.

Comment (cijothomas) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricData.cs:42]: That PR is doing a different (related) thing. Its trying to store count(long) into an int value..
This one is accepting long, and using the built-in conversion to double. It does lose precision, hence i suggested add comment to link to docs, which talks about prevision lose..

Comment (smbecker) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricData.cs:44]: Added a comment and link for reference.

**Review by vishweshbankwar — APPROVED** [review]

**Comment by vishweshbankwar** [comment]
@smbecker - Thanks for your contribution.
